cli/utils.go

```diff
@@ -223,3 +226,128 @@ func replaceLastOccurrence(search string, replacement string, subject string) st
 	}
 	return subject
 }
+
+func getOwnHostname() string {
+	hostname, err := os.Hostname()
+	if err != nil {
+		printWarning("was unable to determine the hostname of the machine")
+		return "unnamed-device"
+	}
+	return hostname
+}
+
+var uuidv4Regex = regexp.MustCompile("^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$")
+
+func looksLikeUuidv4(uuid string) bool {
+	return uuidv4Regex.MatchString(uuid)
+}
+
+var ipv6Regex = regexp.MustCompile("^(?:[0-9a-fA-F]{1,4}:){7}[0-9a-fA-F]{1,4}$")
+
+func looksLikeIpv6(uuid string) bool {
+	return ipv6Regex.MatchString(uuid)
+}
+
+func findGroupUuidByName(needle string, haystack types.Groups) (string, error) {
+	for _, group := range haystack {
+		if group.Name == needle {
+			return group.Id, nil
+		}
+	}
+	return "", fmt.Errorf("Couldn't find a group with name '%s'. Check the spelling.", needle)
+}
```

Comment:
Kinda feels like all `.*(host|*)name` finders should have a second Levenshtein (or something) pass if nothing was found exactly. Can be left as a TODO though